### ISSUE TITLE
curl client disconnects after receiving just one message

### ISSUE THREAD

**Issue by benwilber** [slact/nchan#369]
I am trying to use just a regular curl client to receive messages in pubsub mode, but the client gets disconnected after receiving just the first message.

```nginx
  location ~ ^/chan/(.+)$ {
    expires -1d;
    nchan_pubsub;
    nchan_channel_id $1;
    nchan_channel_id_split_delimiter ",";
    nchan_store_messages off;
  }
```

I subscribe with:

```shell
# curl -vns http://localhost:8080/chan/foo
* Couldn't find host localhost in the .netrc file; using defaults
* About to connect() to localhost port 8080 (#0)
*   Trying ::1...
* Connection refused
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /chan/foo HTTP/1.1
> User-Agent: curl/7.29.0
> Host: localhost:8080
> Accept: */*
> 

```

and publish with:

```shell
# curl -v -X POST -d "Hello World" http://localhost:8080/chan/foo
* About to connect() to localhost port 8080 (#0)
*   Trying ::1...
* Connection refused
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> POST /chan/foo HTTP/1.1
> User-Agent: curl/7.29.0
> Host: localhost:8080
> Accept: */*
> Content-Length: 11
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 11 out of 11 bytes
< HTTP/1.1 201 Created
< Server: nginx/1.10.2
< Date: Mon, 03 Jul 2017 18:43:36 GMT
< Content-Type: text/plain
< Content-Length: 101
< Connection: keep-alive
< Expires: Sun, 02 Jul 2017 18:43:36 GMT
< Cache-Control: no-cache
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: *
< Access-Control-Expose-Headers: Server,range,Content-Length,Content-Range
< Access-Control-Allow-Methods: GET,HEAD,OPTIONS
< 
queued messages: 0
last requested: 58 sec. ago
active subscribers: 0
* Connection #0 to host localhost left intact
last message id: 1499107416:0
```

But the curl subscriber disconnects after receiving just the first message:

```shell
< HTTP/1.1 200 OK
< Server: nginx/1.10.2
< Date: Mon, 03 Jul 2017 18:43:36 GMT
< Content-Type: application/x-www-form-urlencoded
< Content-Length: 11
< Last-Modified: Mon, 03 Jul 2017 18:43:36 GMT
< Connection: keep-alive
< Etag: 0
< Vary: If-None-Match, If-Modified-Since
< Expires: Sun, 02 Jul 2017 18:43:36 GMT
< Cache-Control: no-cache
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: *
< Access-Control-Expose-Headers: Server,range,Content-Length,Content-Range
< Access-Control-Allow-Methods: GET,HEAD,OPTIONS
< 
* Connection #0 to host localhost left intact
Hello World
[root@localhost]#
```

I would like the curl client connection to stay open and receive multiple messages published to the channel.  I was previously using [nginx-push-stream-module](https://github.com/wandenberg/nginx-push-stream-module) and this worked fine, so I'm wondering if there is something different I need to do to achieve this with nchan?

I'm using nginx 1.10.2 and nchan 1.1.6.

Thanks!

**Comment by slact** [comment]
The default HTTP subscriber type is Long-Poll. You can set it to be the equivalent of push-stream-dodule's subscriber type. See https://nchan.io/#http-raw-stream.

**Comment by benwilber** [comment]
@slact thanks for your help.  I seem to be getting a segfault when I change the subscriber type:

```nginx
server {
  listen 8080;
  server_name _;

  location ~ ^/chan/(.+)$ {
    nchan_pubsub http websocket eventsource longpoll chunked multipart-mixed;
    nchan_channel_id $1;
    nchan_channel_id_split_delimiter ",";
    nchan_store_messages off;
  }
}
```

```shell
# nginx -t
Segmentation fault
```

I ran it through strace if that's helpful:

```
open("/etc/nginx/conf.d/nchan.conf", O_RDONLY) = 5
fstat(5, {st_mode=S_IFREG|0644, st_size=252, ...}) = 0
pread(5, "server {\n  listen 8080;\n  server"..., 252, 0) = 252
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x7fe100000002} ---
+++ killed by SIGSEGV +++
Segmentation fault
```

It happens if I add any arguments to `nchan_pubsub` directive at all, instead of just using the defaults.  It even happens if I just explicitly set the defaults.

**Comment by slact** [comment]
Thanks for the bug report, I've just fixed this in ce233c0b. This fix will be in the next release. Meanwhile, you can use the two settings `nchan_publisher` and `nchan_subscriber` together in the same location to the same effect.
